public/app/features/variables/query/reducer.ts

```diff
@@ -86,6 +86,22 @@ const sortVariableValues = (options: any[], sortOrder: VariableSort) => {
   return options;
 };
 
+const getAllMatches = (str: string, regex: RegExp): any => {
+  let results = {};
+  let matches;
+
+  if (regex.global) {
+    while ((matches = regex.exec(str))) {
+      _.merge(results, matches);
+    }
+  } else {
+    if ((matches = regex.exec(str))) {
+      _.merge(results, matches);
+    }
+  }
+  return results;
```

Comment:
This looks ok, but having the nested else doesn't seem that clean. What about something like this
```suggestion
    const results = {};
  let matches;

  do {
    matches = regex.exec(str);
    _.merge(results, matches);
  } while (regex.global && matches);

  return results;
```

Reply:
Agree, much cleaner. Extracting the method was already part of a refactoring. I always need to push a bit further...